### PR TITLE
Add single thread fallback implementation for PrefetchDataset to supp…

### PR DESCRIPTION
**Pull request by boeddeker**
…ort non indexable input

### CODE REVIEW

File: lazy_dataset/core.py [hunk 1]
```diff
@@ -1861,6 +1885,82 @@ def catcher(index):
...
+                for ex in input_dataset:
+                    if shutdown:
+                        return
+                    data_queue.put(ex)
+                    if shutdown:
+                        return
+            finally:
+                data_queue.put(unique_object)
+
+        unique_object = object()
```
Comment (thequilo) [inline lazy_dataset/core.py:1945]: Can you move this definition before `worker`? It is used in `worker`

File: lazy_dataset/core.py [hunk 2]
```diff
@@ -1861,6 +1885,82 @@ def catcher(index):
...
+            while True:
+                item = data_queue.get()
+                if item is unique_object:
+                    break
+                else:
+                    yield item
+        finally:
+            shutdown = True
+            try:
+                data_queue.get_nowait()
```
Comment (thequilo) [inline lazy_dataset/core.py:1959]: Can you add a comment on why this get is necessary?

File: lazy_dataset/core.py (hunk 1 above)
Comment (boeddeker) [reply lazy_dataset/core.py]: Done

File: lazy_dataset/core.py (hunk 2 above)
Comment (boeddeker) [reply lazy_dataset/core.py]: Done

**Review by thequilo — APPROVED** [review]

**Pull request merged**